src/ir/localize.h

```diff
@@ -106,6 +137,36 @@ struct ChildLocalizer {
       }
     }
   }
+
+  // Helper that gets a replacement for the parent: a block containing the
+  // sets + the parent. This will not contain the parent if we don't need it
+  // (if it was never reached).
+  Expression* getReplacement() {
+    if (sets.empty()) {
+      // Nothing to add.
+      return parent;
+    }
+
+    auto* block = Builder(wasm).makeBlock();
+    block->list.set(sets);
+    if (hasUnreachableChild) {
+      // If there is an unreachable child then we do not need the parent at all,
+      // and we know the type is unreachable.
```

Comment:
What if the parent is needed for validation like the post-unreachable children? For example it could be a `br_if` where the condition is unreachable but where it is the only reason its target block has type none.

Reply:
Good point. That is not an issue atm as it is called on things that do not have that behavior, but the user needs to be aware of that. I'll add a note.

Reply:
(added)

Reply:
Would it be safer to just include the parent and let later DCE clean it up?

Reply:
The caller needs to remove this code in some cases, and doing it there would be parsing the output from here to see if it is a block, and then removing checking if there is an unreachable child, and then removing the last block element if so. Then seems more complex to me.